src/predicate_lang/predicate_lang.ml

```diff
@@ -95,47 +95,88 @@ let rec to_dyn f =
   | Union xs -> variant "or" (List.map ~f:(to_dyn f) xs)
   | Inter xs -> variant "and" (List.map ~f:(to_dyn f) xs)
 
-let rec exec t ~standard elem =
-  match (t : _ t) with
-  | Compl t -> not (exec t ~standard elem)
-  | Element f -> elem f
-  | Union xs -> List.exists ~f:(fun t -> exec t ~standard elem) xs
-  | Inter xs -> List.for_all ~f:(fun t -> exec t ~standard elem) xs
-  | Standard -> exec standard ~standard elem
-
-let rec map t ~f =
+let rec test t ~standard ~test:check elem =
```

Comment:
You could define it as `test_` and then have `let test = test_` to avoid renaming things back and force.

Reply:
Done.